crates/proof-of-sql/src/sql/parse/query_context_builder.rs

```diff
@@ -313,9 +313,8 @@ pub(crate) fn type_check_binary_operation(
         BinaryOperator::Add | BinaryOperator::Subtract => {
             try_add_subtract_column_types(*left_dtype, *right_dtype).is_ok()
         }
-        BinaryOperator::Multiply | BinaryOperator::Division => {
-            left_dtype.is_numeric() && right_dtype.is_numeric()
-        }
+        BinaryOperator::Multiply => try_multiply_column_types(*left_dtype, *right_dtype).is_ok(),
+        BinaryOperator::Division => left_dtype.is_numeric() && right_dtype.is_numeric(),
```

Comment:
I believe this is a breaking change.
Did `SELECT a*b from table` work before when `a` and `b` were both `decimal(38,0)`?
Does it work now?
To be clear, I support this change, but it should be marked as breaking if it is.

Reply:
It would have worked in polars. It doesn't work now. Yup.